Source/WebCore/page/Navigation.cpp

```diff
@@ -531,7 +572,8 @@ bool Navigation::innerDispatchNavigateEvent(NavigationNavigationType navigationT
 
     if (event->defaultPrevented()) {
         // FIXME: If navigationType is "traverse", then consume history-action user activation.
-        // FIXME: If event's abort controller's signal is not aborted, then abort the ongoing navigation given navigation.
+        if (!event->signal()->aborted())
+            abortOngoingNavigation(event);
         m_ongoingNavigateEvent = nullptr;
```

Comment:
This is also done by abortOngoingNavigation (indirectly), should it be part of an else section?